tests/rptest/services/compatibility/compat_example.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import sys
+import time
+from ducktape.services.background_thread import BackgroundThreadService
+
+
+class CompatExample(BackgroundThreadService):
+    def __init__(self, context):
+        super(CompatExample, self).__init__(context, num_nodes=1)
+        self._cmds = None
+        self._condition = None
+        self._condition_met = False
+        self._timeout = 10
+        self._node = None
+
+    def _worker(self, idx, node):
+        start_time = time.time()
+        if self._node is None:
+            self._node = node
+
+        #Run the example until the condition is met
+        output_iter = self._node.account.ssh_capture(self._cmds[0])
+        while not self._condition_met and time.time() < start_time + self._timeout:
+            line = next(output_iter)
+            self._condition_met = self._condition(line.rstrip())
```

Comment:
I generally like where this is going. A generic service that is parameterized on a condition. Cool!